siwe.go

```diff
@@ -0,0 +1,462 @@
+			}
+			value = val.(string)
+		default:
+			return nil, &InvalidMessage{"`expirationTime` must be either an ISO8601 formatted string or time.Time"}
+		}
+		expirationTime = &value
+	}
+
+	var notBefore *string
+	if val, ok := options["notBefore"]; ok {
+		var value string
+		switch val.(type) {
+		case time.Time:
+			value = val.(time.Time).UTC().Format(time.RFC3339)
+		case string:
+			_, err := iso8601.ParseString(val.(string))
+			if err != nil {
+				return nil, &InvalidMessage{"Invalid string format for field `notBefore`"}
+			}
+			value = val.(string)
+		default:
+			return nil, &InvalidMessage{"`notBefore` must be either an ISO8601 formatted string or time.Time"}
+		}
+		notBefore = &value
+	}
+
+	var requestID *string
+	if val, ok := options["requestId"]; ok {
+		value := val.(string)
+		requestID = &value
+	}
+
+	var resources []string
+	if val, ok := options["resources"]; ok {
+		switch val.(type) {
+		case []string:
+			resources = val.([]string)
+		default:
+			return nil, &InvalidMessage{"`resources` must be a []string"}
+		}
+	}
+
+	return &Message{
+		domain:  domain,
+		address: common.HexToAddress(address),
+		uri:     uri,
+		version: version,
+
+		statement: statement,
+		nonce:     nonce,
+		chainID:   chainId,
+
+		issuedAt:       issuedAt,
+		expirationTime: expirationTime,
+		notBefore:      notBefore,
+
+		requestID: requestID,
+		resources: resources,
+	}, nil
+}
+
+func (m *Message) GetDomain() string {
+	return m.domain
+}
+
+func (m *Message) GetAddress() common.Address {
+	return m.address
+}
+
+func (m *Message) GetURI() string {
+	return m.uri
+}
+
+func (m *Message) GetVersion() string {
+	return m.version
+}
+
+func (m *Message) GetStatement() *string {
+	if m.statement != nil {
+		ret := *m.statement
+		return &ret
+	}
+	return nil
+}
+
+func (m *Message) GetNonce() *string {
+	if m.nonce != nil {
+		ret := *m.nonce
+		return &ret
+	}
+	return nil
+}
+
+func (m *Message) GetChainID() string {
+	return m.chainID
+}
+
+func (m *Message) GetIssuedAt() string {
+	return m.issuedAt
+}
+
+func (m *Message) getExpirationTime() *time.Time {
+	if !isEmpty(m.expirationTime) {
+		ret, _ := iso8601.ParseString(*m.expirationTime)
+		return &ret
+	}
+	return nil
+}
+
+func (m *Message) GetExpirationTime() *string {
+	if m.expirationTime != nil {
+		ret := *m.expirationTime
+		return &ret
+	}
+	return nil
+}
+
+func (m *Message) getNotBefore() *time.Time {
+	if !isEmpty(m.notBefore) {
+		ret, _ := iso8601.ParseString(*m.notBefore)
+		return &ret
+	}
+	return nil
+}
+
+func (m *Message) GetNotBefore() *string {
+	if m.notBefore != nil {
+		ret := *m.notBefore
+		return &ret
+	}
+	return nil
+}
+
+func (m *Message) GetRequestID() *string {
+	if m.requestID != nil {
+		ret := *m.requestID
+		return &ret
+	}
+	return nil
+}
+
+func (m *Message) GetResources() []string {
+	return m.resources
+}
+
+func GenerateNonce() string {
+	return uniuri.NewLen(16)
```

Comment:
That package generates random strings for uniqueness, not cryptographically sound nonces

Reply:
I'm looking for another alternative, but it seems to be basically the same thing we do in siwe-rs .
```
    thread_rng()
        .sample_iter(&Alphanumeric)
        .take(11)
        .map(char::from)
        .collect()
```

Reply:
My bad, it looks sound, I got confused with their focus on UUIDs